### PR TITLE
Clinical submission upload reopen approve

### PR DESCRIPTION
**Pull request by UmmulkiramR**
contains:
- clinical submission upload
- reopen 
- approve
- tests

### CODE REVIEW

File: src/schemas/utils.ts [hunk 1]
```diff
@@ -170,34 +172,42 @@ export interface SubmissionEntity {
 }
 
 const convertClinicalSubmissionEntityToGql = (clinicalType: string, entity: SubmissionEntity) => {
+  const batchName = entity.batchName || undefined;
```
Comment (demariadaniel) [inline src/schemas/utils.ts:175]: This and other declarations can use optional chaining:
`const batchName = entity?.batchName;`

File: test/integration/submission/submission.spec.ts [hunk 2]
```diff
@@ -850,8 +854,9 @@ describe('Submission Api', () => {
       } catch (err) {}
       await chai
         .request(app)
-        // data base is empty so ID shouldn't exist
-        .post('/submission/program/ABCD-EF/clinical/upload')
+        // database is empty so ID shouldn't exist
+        // .post('/submission/program/ABCD-EF/clinical/upload')
```
Comment (demariadaniel) [inline test/integration/submission/submission.spec.ts:858]: If these commented lines are unused we should remove them

However, the bigger problem is that we now have two endpoints, `/upload` and `/submissionUpload`
If we deprecate `/upload` we have to make a change on the Gateway so existing functionality is not broken

And we should keep unit tests for `/upload` unless it is being removed... I'm wondering if we should we have separate unit tests for `/upload` and `/submissionUpload` at this stage

File: src/schemas/clinical-mutations/uploadClinicalSubmissions.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+async function uploadClinicalSubmissions(req: Request, res: Response) {
+  const programShortName = req.params.programId;
+  const clinicalFiles = req.files as Express.Multer.File[];
+
+  const egoToken = (req.headers.authorization || '').split('Bearer ').join('');
+  const permissionsFromToken = egoTokenUtils.getPermissionsFromToken(egoToken);
+
+  /*if (!egoTokenUtils.canWriteSomeProgramData(permissionsFromToken)) {
+    throw new AuthenticationError('User is not authorized to write data');
+  }*/
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/schemas/clinical-mutations/uploadClinicalSubmissions.ts:35]: Is `permissionsFromToken` being used?

File: test/integration/submission/submission.spec.ts (hunk 2 above)
Comment (demariadaniel) [reply test/integration/submission/submission.spec.ts:858]: Discussed on Slack -- plan to update after migration to Cumulus

File: src/schemas/clinical-mutations/uploadClinicalSubmissions.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (UmmulkiramR) [reply src/schemas/clinical-mutations/uploadClinicalSubmissions.ts]: removed it as we are handling the permissions check in subbmissionApi

File: src/schemas/utils.ts (hunk 1 above)
Comment (UmmulkiramR) [reply src/schemas/utils.ts]: done.

**Review by demariadaniel — APPROVED** [review]

**Pull request merged**